timestamp-impl/src/main/java/com/palantir/timestamp/PersistentUpperLimit.java

```diff
@@ -53,8 +58,10 @@ public long get() {
     }
 
     public synchronized void increaseToAtLeast(long minimum) {
-        if(cachedValue < minimum) {
+        if (cachedValue < minimum) {
             store(minimum);
+        } else {
+            log.trace("Not storing upper limit of {}, as the cached value {} was higher.", minimum, cachedValue);
```

Comment:
Yeah, though I want to be sure something is not getting a racy read/write or even less likely hitting a JMM bug.